src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -234,7 +234,7 @@ public static class SymbolDisplay
             // and, for consistency, with other display options.
             if ((symbol as Symbols.PublicModel.MethodSymbol)?.UnderlyingMethodSymbol is SynthesizedSimpleProgramEntryPointSymbol)
             {
-                return ImmutableArray.Create<SymbolDisplayPart>(new SymbolDisplayPart(SymbolDisplayPartKind.MethodName, symbol, "<simple-program-entry-point>"));
+                return ImmutableArray.Create<SymbolDisplayPart>(new SymbolDisplayPart(SymbolDisplayPartKind.MethodName, symbol, "<top-level-statements-entry-point>"));
```

Comment:
><top-level-statements-entry-point> [](start = 129, length = 34)

Consider "<entry-point>" or better yet avoid the special case so this is reported as "$Program.$Main".

Reply:
I believe there is a value in providing more specific information here. I also believe that exposing compiler generated unspeakable names is bad. In the specification we intentionally doe not specify the names and without the special case the name will leak into error messages, etc.

---
In reply to: [426755307](https://github.com/dotnet/roslyn/pull/44305#discussion_r426755307) [](ancestors = 426755307)